spec/codecs/protobuf3_spec.rb

```diff
@@ -0,0 +1,269 @@
+# encoding: utf-8
+require "logstash/devutils/rspec/spec_helper"
+require "logstash/codecs/protobuf"
+require "logstash/event"
+require "insist"
+
+
+require 'google/protobuf' # for protobuf3
+
+
+describe LogStash::Codecs::Protobuf do
+
+  pb_include_path = "../../../spec/helpers/"
+
+  context "#decodePB3" do
+
+
+    #### Test case 1: Decode simple protobuf ####################################################################################################################
+    let(:plugin_unicorn) { LogStash::Codecs::Protobuf.new("class_name" => "Unicorn", "include_path" => [pb_include_path + '/pb3/unicorn_pb.rb'], "protobuf_version" => 3)  }
```

Comment:
I suggest wrapping each test case in its own `describe` or `context` block. This way each `register` done in a  `before` won't impact another test.

Right now, because all `before` blocks are at the same level, they all run for each test. This is why a failure to compile `plugin_5` is causing 5 tests to fail instead of just one.

Reply:
Sounds good! I will apply that change later today. Thank you so much!

Reply:
Done. Sorry for the delay :)